src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Source.cs

```diff
@@ -25,17 +29,45 @@ private static void FreeSymbolMap(MultiDictionary<string, ISymbol> symbolMap)
         }
 
         public static async Task<SymbolTreeInfo> GetInfoForSourceAssemblyAsync(
-            Project project, CancellationToken cancellationToken)
+            Project project, Checksum checksum, CancellationToken cancellationToken)
         {
             var compilation = await project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
 
             return await LoadOrCreateSourceSymbolTreeInfoAsync(
-                project.Solution, compilation.Assembly, project.FilePath,
+                project.Solution, compilation.Assembly, checksum, project.FilePath,
                 loadOnly: false, cancellationToken: cancellationToken).ConfigureAwait(false);
         }
 
+        public static async Task<Checksum> GetSourceSymbolsChecksumAsync(Project project, CancellationToken cancellationToken)
+        {
+            // The SymbolTree for source is built from the source-symbols from the project's compilation's
+            // assembly.  Specifically, we only get the name, kind and parent/child relationship of all the
+            // child symbols.  So we want to be able to reuse the index as long as none of these have 
+            // changed.  The only thing that can make those source-symbols change in that manner are if
+            // the text of any document changes, or if options for the project change.  So we build our
+            // checksum out of that data.
+            var serializer = new Serializer(project.Solution.Workspace);
+
+            var textChecksumsTasks = project.Documents.Select(async d =>
+            {
+                var text = await d.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                return serializer.CreateChecksum(text, cancellationToken);
+            });
+
+            var textChecksums = await Task.WhenAll(textChecksumsTasks).ConfigureAwait(false);
```

Comment:
this seems can be moved after other two checksum. these are nit so you can ignore if don't want to update PR.